collab-service/src/message_handler/internal/internal_message_types.d.ts

```diff
@@ -0,0 +1,25 @@
+const enum ConnectionFlag {
+  NONE,
+  JOIN,
+  ACK,
+}
+
+declare type TunnelMessage = {
+  sender: string,
+  nick: string,
```

Comment:
Is there a need to introduce the `nick` field? You can just encode the nickname in data and it'll semantically make sense.

Reply:
I suppose it is not necessary. I'll move it over to the data since the join & ack messages have are transferring empty data